stdlib/src/builtin/float_literal.mojo

```diff
@@ -221,7 +222,61 @@ struct FloatLiteral(
             return truncated
         return truncated + 1
 
-    # TODO: implement __round__
+    @always_inline("nodebug")
+    fn __round__(self) -> Self:
+        """Return the rounded value of the FloatLiteral.
+
+        Returns:
+            The rounded value.
+        """
+        # Handle special values first.
+        if not self._is_normal():
+            return self
+
+        var truncated: IntLiteral = self.__int_literal__()
+        var result: Self
+        if abs(self) - abs(truncated) <= 0.5:
+            result = Self(truncated)
+        elif self > 0:
+            result = Self(truncated + 1)
+        else:
+            result = Self(truncated - 1)
+        return result
+
+    @always_inline("nodebug")
+    fn __round__(self, ndigits: IntLiteral) -> Self:
+        """Return the rounded value of the FloatLiteral.
+
+        Args:
+            ndigits: The number of digits to round to. Defaults to 0.
+
+        Returns:
+            The rounded value.
+        """
+        # Handle special values first.
+        if not self._is_normal():
+            return self
+
+        alias one = __mlir_attr.`#kgen.int_literal<1> : !kgen.int_literal`
+        alias ten = __mlir_attr.`#kgen.int_literal<10> : !kgen.int_literal`
+        var multiplier = one
+        # TODO: Use IntLiteral.__pow__() when it's implemented.
+        for _ in range(ndigits):
+            multiplier = __mlir_op.`kgen.int_literal.binop`[
+                oper = __mlir_attr.`#kgen<int_literal.binop_kind mul>`
+            ](multiplier, ten)
+        var target: Self = self * Self(multiplier)
+        var truncated: IntLiteral = target.__int_literal__()
```

Comment:
To be defensive here (because implicit conversions can cause some unintuitive behavior below) and to decrease dependency on `IntLiteral`, can you just declare this as `Self`, and convert right away? Then you don't need the explicit casts to `Self` below.

Reply:
Sure, done: https://github.com/msaelices/mojo/commit/f879b9b54b701fe4762de1123f3a4fe64ab40348